### PR TITLE
whatsnew entry for dependencies

### PR DESCRIPTION
**Pull request by bjlittle**
This PR adds the `whatsnew` entry for #3812.

It also addresses several minor documentation issues that have unfortunately crept in :thinking: 

We need to fix the `sphinx` logger issue #3789

### CODE REVIEW

File: lib/iris/common/metadata.py [hunk 1]
```diff
@@ -1198,7 +1198,7 @@ def _check(item):
 
 class DimCoordMetadata(CoordMetadata):
     """
-    Metadata container for a :class:`~iris.coords.DimCoord"
+    Metadata container for a :class:`~iris.coords.DimCoord`
```
Comment (bjlittle) [inline lib/iris/common/metadata.py:1201]: @tkknight This was just wrong, and causing a documentation warning during `sphinx` build time.

File: docs/iris/src/whatsnew/latest.rst [hunk 2]
```diff
@@ -35,16 +35,17 @@ Features
...
 ==================
 
-* Iris now supports the latest version of
-  `Proj <https://github.com/OSGeo/PROJ>`_.
+* Iris now supports the latest version of `Proj <https://github.com/OSGeo/PROJ>`_.
 
 * Iris now requires `Cartopy <https://github.com/SciTools/cartopy>`_ >= 0.18 in
-  order to remain compatible with the latest version of
-  `Matplotlib <https://github.com/matplotlib/matplotlib>`_.
+  order to remain compatible with the latest version of `Matplotlib`_.
```
Comment (bjlittle) [inline docs/iris/src/whatsnew/latest.rst:41]: @tkknight There ended up being **two** `matplotlib` references...

File: docs/iris/src/userguide/regridding_plots/regridding_plot.py [hunk 3]
```diff
@@ -22,6 +22,4 @@
 ax.coastlines(resolution="50m")
 ax.gridlines()
 
-plt.tight_layout()
-
```
Comment (bjlittle) [inline docs/iris/src/userguide/regridding_plots/regridding_plot.py:26]: @tkknight The `plt.tight_layout()` was causing a traceback within `matplotlib`, resulting in the plot not being rendered in the documentation i.e., see [here](https://scitools-iris.readthedocs.io/en/latest/userguide/interpolation_and_regridding.html#regridding).

**Comment by tkknight** [comment]
Looks good @bjlittle.

**Pull request merged**